server.ts

```diff
@@ -815,8 +865,13 @@ app.post('/job/:jobId/submit', async function (req, res) {
             .where('id = :id', { id:  job.id })
             .set({ queuedAt: job.queuedAt })
             .execute()
-        if (job.hpc == "dummy_hpc") {
+        if (job.hpc == "instant_hpc") {
             job.finishedAt = new Date()
+            await connection.createQueryBuilder()
+            .update(Job)
```

Comment:
Saving the same Job twice is not necessary. You only need to update the job once. Just set the finishedAt to current date (new Date) at the line before `if (job.hpc == "instant_hpc")` if its `instant_hpc `

Reply:
Does this look good now ?